users/views.py

```diff
@@ -125,29 +135,41 @@ def form_valid(self, form: SimpleUserChangeForm) -> HttpResponse:
 
 @method_decorator(login_required, name="dispatch")
 @method_decorator(check_permissions(allowed_user_types=[CustomUser.UserType.ADMIN.name]), name="dispatch")
-class UserUpdateByAdmin(UpdateView):
-    template_name = "user_update.html"
+class UserUpdateByAdmin(BSModalUpdateView):
+    template_name = "modals/create_and_update_user.html"
     form_class = AdminUserChangeForm
     context_object_name = "user_detail"
     model = CustomUser
 
     def get_success_url(self) -> str:
-        return reverse("custom-user-update-by-admin", kwargs={"pk": self.object.pk})
+        return reverse("custom-users-list")
 
     def form_valid(self, form: AdminUserChangeForm) -> HttpResponse:
         super().form_valid(form)
         logger.info(f"User with id: {self.object.pk} has been updated by admin with id {self.request.user.pk}")
         messages.success(self.request, ConfirmationMessages.SUCCESSFUL_UPDATE_USER_MESSAGE)
         return redirect(self.get_success_url())
 
+    def get_context_data(self, *, _object_list: Any = None, **kwargs: Any) -> dict:
+        context_data = super().get_context_data(**kwargs)
+        context_data["UI_text"] = UserModalText
+        context_data["UI_text_common"] = UsersListText
+        context_data["is_create_user_form"] = False
+        return context_data
+
 
 @method_decorator(login_required, name="dispatch")
 @method_decorator(check_permissions(allowed_user_types=[CustomUser.UserType.ADMIN.name]), name="dispatch")
 class UserList(ListView):
     template_name = "users_list.html"
     model = CustomUser
-    queryset = (
-        CustomUser.objects.active()
+    queryset_active = (
+        CustomUser.objects.filter(is_active=True)
+        .order_by("user_type", "last_name", "first_name", "email")
+        .prefetch_related("projects")
+    )
+    queryset_disabled = (
+        CustomUser.objects.filter(is_active=False)
```

Comment:
Instead of two database calls you can do here only one:
``python
ordered_users_queryset = (
    CustomUser.objects
    .order_by("user_type", "last_name", "first_name", "email")
    .prefetch_related("projects")
)

disabled_users_queryset = ordered_users_queryset.filter(is_active=False)
active_users_queryset = ordered_users_queryset.filter(is_active=True)
```
It should only once call database and also should be properly ordered. Please ensure that proper ordering is done in this case.

Reply:
Done